docs/_pages/grammar.md

```diff
@@ -60,19 +58,25 @@ unop = '-' | 'not' | '#'
 
 SimpleType =
     'nil' |
-    NAME ['.' NAME] [ '<' TypeList '>' ] |
+    SingletonType |
+    NAME ['.' NAME] [ '<' [TypeParams] '>' ] |
     'typeof' '(' exp ')' |
     TableType |
     FunctionType
 
+SingletonType = STRING | 'true' | 'false'
+
 Type =
     SimpleType ['?'] |
     SimpleType ['|' Type] |
     SimpleType ['&' Type]
 
-GenericTypeList = NAME ['...'] {',' NAME ['...']}
+GenericTypePack = NAME '...' ['=' '(' TypeList ')']
```

Comment:
I believe we also permit default type pack values that refer to a type pack like `...number`; see https://github.com/Roblox/luau/blob/master/rfcs/syntax-default-type-alias-type-parameters.md#design